src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -1350,6 +1374,7 @@ public FormValidation doValidate(
                 @QueryParameter int readTimeout,
                 @QueryParameter int threadExecutorNumber,
                 @QueryParameter boolean useBearerAuth,
+                @QueryParameter int maxIssuesFromJqlSearch,
```

Comment:
I _think_ this might not be needed and it should be handled automatically by Jelly [validation as described here](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/lib/form/number.jelly#L62) but I haven't tested that, can you give it a try? That would avoid a lot of changes.

Reply:
It is not working, if I remove this parameter. 

on click of Validate Settings button, Success message is shown even if the number is bigger than 5000. 

![grafik](https://github.com/user-attachments/assets/77280b2d-91f4-4202-93c6-a43794dc9994)